src/app/config.rs

```diff
@@ -40,11 +40,21 @@ pub struct Config {
     /// The maximum amount of time to wait for a connection to a remote peer.
     pub outbound_connect_timeout: Duration,
 
-    /// The amount of time to wait between connection attempts.
-    pub inbound_connect_backoff: Duration,
+    /// Settings for the exponential back-off used to determine the time the
+    /// amount of time to wait between connection attempts.
+    pub inbound_connect_exp_backoff_min: Duration,
 
-    /// The amount of time to wait between connection attempts.
-    pub outbound_connect_backoff: Duration,
+    pub inbound_connect_exp_backoff_max: Duration,
+
+    pub inbound_connect_exp_backofff_max_jitter: f64,
+
+    /// Settings for the exponential back-off used to determine the time the
+    /// amount of time to wait between connection attempts.
+    pub outbound_connect_exp_backoff_min: Duration,
+
+    pub outbound_connect_exp_backoff_max: Duration,
+
+    pub outbound_connect_exp_backofff_max_jitter: f64,
```

Comment:
Take it or leave it: we're slowly in the process of "un-flattening" this large config structure... What do you think about defining a `struct ExpBackoff` in this module, and then we can simply store `outbound_connect_backoff: ExpBackoff` in the config struct.

In fact... what do you think about making `reconnect`'s `Backoff` struct public? Then we could just store that structure here; and the reconnect module could be changed to have a `with_backoff(Backoff)`...